CRM/Mailing/BAO/Mailing.php

```diff
@@ -1482,15 +1486,16 @@ public static function add(&$params, $ids = array()) {
    * @throws \Exception
    */
   public static function create(&$params, $ids = array()) {
-    // WTH $ids
-    if (empty($ids) && isset($params['id'])) {
-      $ids['mailing_id'] = $ids['id'] = $params['id'];
+
+    if (empty($params['id']) && !empty($ids)) {
+      $params['id'] = isset($ids['mailing_id']) ? $ids['mailing_id'] : $ids['id'];
```

Comment:
@eileenmcnaughton getting error of undefined index `id` here

Reply:
thanks - I think I got it this time